File: clojure/armstrong-numbers/src/armstrong_numbers.clj
(ns armstrong-numbers)

(def expt #(reduce * (repeat %1 %2)))
(def parse-int #(Integer/parseInt %))
(def split-digits #(map (comp parse-int str) (str %)))

(defn armstrong? [num]
  (let [digits (split-digits num)
        length (count digits)]
    (->>
     digits
     (map #(expt length %)) ;; raise each digit to "length"
     (reduce +)             ;; sum them to get the armstrong-sum
     (= num))))             ;; check if armstrong-sum == number

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"math"
)

func IsNumber(n int) bool {
	return n == armstrongSum(n)
}

func armstrongSum(n int) int {
	var sum int
	nums := digits(n)

	for _, n := range nums {
		sum += pow(n, len(nums))
	}

	return sum
}

func digits(n int) []int {
	digits := []int{}

	for r := n; r > 0; r /= 10 {
		digits = append(digits, r%10)
	}

	return digits
}

func pow(a, b int) int {
	return int(math.Pow(float64(a), float64(b)))
}
